### DISCUSSION TITLE
Add gizmos to NavigationLink3D's start and end position

### DISCUSSION THREAD

**Discussion by venilark** [godotengine/godot-proposals#6099] [3D]
Moving the start and end position of a NavigationLink3D is kinda hard right now, which would be solved with help of gizmos like in any node that extends from Node3D (like the NavigationLink3D itself).

![nl](https://user-images.githubusercontent.com/98270528/212476847-8e63c754-6578-4230-8b78-f64c5969106e.PNG)

**Comment by smix8** [comment]
@DarkKilauea

I remember we discussed this but it is not easy to add proper sub gizmos for nodes that require the Node3DGizmo.

**Reply by venilark** [reply]
> @DarkKilauea
> 
> I remember we discussed this but it is not easy to add proper sub gizmos for nodes that require the Node3DGizmo.

what if End and Start became their own nodes, NavigationLink3DStart and NavigationLink3DEnd?
then we could place them anywhere easily and if they don't need the NavigationLink3D parent node, each end could be put as a child of any Node, like a moving platform (in this case I don't know if recalculating the travel cost would require rebaking or not), while still sharing the connection
Start would have an exported NavigationLink3Dend variable and all the info NavigationLink3D currently has

I never tried it and could be wrong but I think Unity lets you add an off-mesh link component to a Transform and then you choose two other transforms as Start and End, this should be similar

**Reply by smix8** [reply]
Adding new nodes is full of trouble and a maintaince burden. It is something that you can do in your own project as a quick solution with a tool script, e.g. having a Node3D that transfers its transform to the JumpLink3D but this is not a solution for the Editor. 

This needs to be fixed by Editor ppl where the mess is created which is inside the Node3DGizmo, cause it is not just NavigationLink3D, all the 3d nodes that need to work with "sub" positions suffer from very poor editor handles, e.g. Path3D, CollisionPolygon3D,

**Reply by DarkKilauea** [reply]
@venilark Hey, sorry for missing this, it's been a crazy month for me so far.

I originally planned to follow a model like [path_corner](https://developer.valvesoftware.com/wiki/Path_corner), where each link node represents a point along a path, with links being created on the server as nodes are added to the chain.  However, Godot does not seem to have a great way to tell if the transform of another node changes, so that idea would require linking together specific types of nodes that could report back that they changed.

Talking to @smix8 in the contributors chat, we discussed using a couple of `NodePath` properties that specify which nodes define the start and end points of the link, then have the `NavigationLink` nodes track the transforms of those nodes and update the link.  The trick is finding a way to do this efficiently.  However, using other nodes, like `Position3D`, would solve the gizmo issue.

It's something I'd like to see more discussion on and more information on how links are being used today.

**Reply by DarkKilauea** [reply]
I've created a prototype using `NodePath` to set the start and end of a link: https://github.com/godotengine/godot/pull/74097

@venilark please take a look and let me know if it addresses your concern.

**Reply by venilark** [reply]
@DarkKilauea I checked this out and added a comment on your PR
